From 5909cd63f0c798e8295511fa9c168f2dbae00e19 Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Tue, 2 Oct 2012 15:46:05 +0000
Subject: [PATCH 0351/1077] LogService.exe; massaiSNMP.dll V1.0.3.27 Mantis
 0024647: LogService writes 00000000-00:00:00 as timestamp in and svc.log,
 platform.log

git-svn-id: svn://localhost/SelfServiceCommon/trunk@417 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/cpp/LogService/inc/LogService_def.h         |  9 ++++++---
 .../Massai/cpp/LogService/inc/massaiSNMP_def.h         |  9 ++++++---
 .../Massai/cpp/LogService/src/logFile.cpp              | 10 ++++++----
 3 files changed, 18 insertions(+), 10 deletions(-)

diff --git a/SelfServiceCommon/Massai/cpp/LogService/inc/LogService_def.h b/SelfServiceCommon/Massai/cpp/LogService/inc/LogService_def.h
index 52c9b709..050e06e8 100644
--- a/SelfServiceCommon/Massai/cpp/LogService/inc/LogService_def.h
+++ b/SelfServiceCommon/Massai/cpp/LogService/inc/LogService_def.h
@@ -11,8 +11,8 @@
 
 #include "Tools/VERSION.H"
 
-#define FILE_VER_BIN     1,0,3,26
-#define FILE_VER_STR     "FileVersion", "1.0.3.26\0"
+#define FILE_VER_BIN     1,0,3,27
+#define FILE_VER_STR     "FileVersion", "1.0.3.27\0"
 
 /*! \page LogService.exe - Logging Service
  *                         writes logging\svc.log logging\platform*.log logging\statistics*.log
@@ -21,10 +21,13 @@
  *
  *  !!! immer auch wenn nötig massaiSNMP_def.h pflegen !!!
  *
- *  @version 1.0.3.26
+ *  @version 1.0.3.27
  *
  *  \section change1 Change history
  *
+ *  \date 02.10.2012 \li V1.0.3.27
+ *                   \li bugfix - SteT Mantis 0024647: LogService writes 00000000-00:00:00 as timestamp in and svc.log, platform.log
+ *
  *  \date 07.09.2012 \li V1.0.3.26
  *                   \li bugfix - Mantis 0021005: LogService.exe: hobbit connection; write always
  *
diff --git a/SelfServiceCommon/Massai/cpp/LogService/inc/massaiSNMP_def.h b/SelfServiceCommon/Massai/cpp/LogService/inc/massaiSNMP_def.h
index ad072c84..cc9bfe14 100644
--- a/SelfServiceCommon/Massai/cpp/LogService/inc/massaiSNMP_def.h
+++ b/SelfServiceCommon/Massai/cpp/LogService/inc/massaiSNMP_def.h
@@ -11,8 +11,8 @@
 
 #include "Tools/VERSION.H"
 
-#define FILE_VER_BIN     1,0,3,26
-#define FILE_VER_STR     "FileVersion", "1.0.3.26\0"
+#define FILE_VER_BIN     1,0,3,27
+#define FILE_VER_STR     "FileVersion", "1.0.3.27\0"
 
 /*! \page massaiSNMP.dll - plugin for MS SNMP-Service;
  *                         writes logging\svc.log logging\platform*.log logging\statistics*.log;
@@ -21,10 +21,13 @@
  *
  *  !!! immer auch wenn nötig logService_def.h pflegen !!!
  *
- *  @version 1.0.3.26
+ *  @version 1.0.3.27
  *
  *  \section change1 Change history
  *
+ *  \date 02.10.2012 \li V1.0.3.27
+ *                   \li bugfix - SteT Mantis 0024647: LogService writes 00000000-00:00:00 as timestamp in and svc.log, platform.log
+ *
  *  \date 07.09.2012 \li V1.0.3.26
  *                   \li bugfix - Mantis 0021005: LogService.exe: hobbit connection; write always
  *
diff --git a/SelfServiceCommon/Massai/cpp/LogService/src/logFile.cpp b/SelfServiceCommon/Massai/cpp/LogService/src/logFile.cpp
index 86cb8fdb..2576e2ce 100644
--- a/SelfServiceCommon/Massai/cpp/LogService/src/logFile.cpp
+++ b/SelfServiceCommon/Massai/cpp/LogService/src/logFile.cpp
@@ -69,7 +69,6 @@ logFile::~logFile()
 
 std::string logFile::getOutputPath(const std::string path)
 {
-	char *pStr;
 	int  iLen;
 	char szOutputPathTmp[328];
 
@@ -220,12 +219,15 @@ void logFile::write(const long timeStamp, const std::string& Text)
 		char timestring[48];
 		struct tm* pTime;
 
-		pTime = localtime( reinterpret_cast< const time_t*>(&timeStamp));
+  time_t ts = timeStamp;
+		pTime = ::localtime(&ts);
 		if(pTime)
-			strftime(timestring, sizeof(timestring), "%Y%m%d-%H:%M:%S", pTime);
+  {
+			 strftime(timestring, sizeof(timestring), "%Y%m%d-%H:%M:%S", pTime);
+  }
 		else
 		{
-			sprintf(timestring,"00000000-00:00:00");
+ 			sprintf(timestring,"00000000-00:00:00");
 		}
 
 		openFile();
-- 
2.41.0.windows.1

